control-plane-operator/controllers/hostedcontrolplane/v2/dnsoperator/deployment.go

```diff
@@ -0,0 +1,78 @@
+package dnsoperator
+
+import (
+	"fmt"
+
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/util"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func adaptDeployment(cpContext component.WorkloadContext, obj *appsv1.Deployment) error {
+	util.UpdateContainer("dns-operator", obj.Spec.Template.Spec.Containers, func(c *corev1.Container) {
+		// TODO (alberto): enforce ImagePullPolicy in component defaults.
+		c.ImagePullPolicy = corev1.PullIfNotPresent
+		c.Command = []string{"dns-operator"}
+		c.Env = []corev1.EnvVar{
+			{
+				Name:  "RELEASE_VERSION",
+				Value: cpContext.ReleaseImageProvider.Version(),
+			}, {
+				Name:  "IMAGE",
+				Value: cpContext.ReleaseImageProvider.GetImage("coredns"),
+			}, {
+				Name:  "OPENSHIFT_CLI_IMAGE",
+				Value: cpContext.ReleaseImageProvider.GetImage("cli"),
+			}, {
+				Name:  "KUBE_RBAC_PROXY_IMAGE",
+				Value: cpContext.ReleaseImageProvider.GetImage("kube-rbac-proxy"),
+			}, {
+				Name:  "KUBECONFIG",
+				Value: "/etc/kubernetes/kubeconfig",
+			},
+		}
+		c.Resources = corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("10m"),
+				corev1.ResourceMemory: resource.MustParse("29Mi"),
+			},
+		}
+		c.TerminationMessagePolicy = corev1.TerminationMessageFallbackToLogsOnError
+		c.VolumeMounts = []corev1.VolumeMount{{
+			Name:      "kubeconfig",
+			MountPath: "/etc/kubernetes",
+		}}
+	})
+	obj.Spec.Template.Spec.TerminationGracePeriodSeconds = ptr.To[int64](2)
+	obj.Spec.Template.Spec.Volumes = []corev1.Volume{{
+		Name: "kubeconfig",
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  "dns-operator-kubeconfig",
+				DefaultMode: ptr.To[int32](0640),
+			},
+		},
+	}}
```

Comment:
all this is already a static part of the `deployment.yaml` manifest, we don't need to set it here again.